### DISCUSSION TITLE
F_CPU is defined but looks like it is not in use

### DISCUSSION THREAD

**Discussion by lyusupov** [SiliconLabs/arduino#79] [Q&A] [answered]
Does it mean that the CPU frequency is hardcoded in each 'variant' SDK library file and can not be changed to a lower MHZ value ?

<img width="877" alt="image" src="https://github.com/user-attachments/assets/a8afbc0e-7f73-405b-bcda-ab1a819cc75a">

**Comment by silabs-bozont** [comment, marked as answer]
Hello @lyusupov,
Your observation is correct - it's not used anywhere in the code. It's only defined for compatibility reasons - we might remove this in the future.
We have a separate API for changing the CPU clock speed:
`setCPUClock() - sets the CPU clock speed - it can be one of CPU_39MHZ, CPU_76MHZ, CPU_80MHZ`
Note, that the radio variants are calibrated to work with a 39 MHz clock.
